module_5/json_function.js

```diff
@@ -0,0 +1,90 @@
+const array = ['Mahi', 2, user];
+console.log(array); // [
+                    //   'Mahi',
+                    //    2,
+                    //    { name: 'john', password: 'john@22', email: 'john@gmail.com' }
+                    //  ] => console the array
+console.log(JSON.stringify(array)); // it stringify the array elements
+                                    // [ 
+                                    //   "Mahi",
+                                    //   2,
+                                    //   {
+                                    //     "name":"john",
+                                    //     "password":"john@22",
+                                    //     "email":"john@gmail.com"
+                                    //   }
+                                    // ]
+
+
+let birthDate = {
+  day: 25,
+  month: 'June',
+  year: 1994
+}
+let address = {
+  lane: 'Shahu road',
+  city: 'Malegaon'
+}
+let myProfile = {
+  name: 'Julee',
+  birthDate: birthDate,
+  address: address
+}  // nested object
+console.log(myProfile); // {
+                        //   name: 'Julee',
+                        //   birthDate: { 
+                        //                   day: 25, 
+                        //                   month: 'June', 
+                        //                   year: 1994 },
+                        //   address: { 
+                        //                 lane: 'Shahu road', 
+                        //                 city: 'Malegaon'
+                        //               }
+                        //  }
+
+console.log(JSON.stringify(myProfile)); // It will stringify the whole obj along with objects within it
+                                        // {
+                                        //   "name":"Julee",
+                                        //   "birthDate":{
+                                        //                  "day":25,
+                                        //                  "month":"June",
+                                        //                  "year":1994
+                                        //                },
+                                        //   "address":{
+                                        //                "lane":"Shahu road",
+                                        //                 "city":"Malegaon"
+                                        //              }
+                                        // }
+// What happens if we pass numbers, strings, undefined, null to JSON.stringify?
+let num = 12;
+let str = 'Josh Software Company';
+let unDef = undefined;
+let nullValue = null;
+console.log(JSON.stringify(num)); // 12
+console.log(JSON.stringify(str)); // "Josh Software Company"
+console.log(JSON.stringify(unDef)); // undefined
+console.log(JSON.stringify(nullValue)); // null
+
+// Conclusion:
+// We can only stringify the string only . We cannot stringify the number or undefined or null values
```

Comment:
If we use number into JSON it converts Number into String data type, for null also it's convert into string but not for undefined Please note this